# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ahmedfurqan1/FlyRank-MachineLearning/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

###Answer
**Lane: Lane 2 — Refresh / Content Opportunity Scoring**

Content search traffic decays over time, but editing teams have limited capacity to audit every URL. Lane 2 builds a ranking pipeline with clear reason codes so teams can prioritize declining, high-demand pages first instead of inspecting pages manually.

---



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

###Answer
**Decision Improved:**
This work improves the decision of which declining web pages should be prioritized for a content review and refresh during the next planning cycle.

**Who Acts:**
SEO managers and content editors act on these predictions. They update outdated information, rewrite meta titles and descriptions, and fix search intent gaps on the flagged pages.

**Cost of a Wrong Call:**
A wrong recommendation carries costs on both sides. A **false positive** wastes writer time and budget editing a healthy page that was already performing well. A **false negative** leaves high-visibility pages unnoticed, causing a quiet loss of organic search traffic and conversion revenue.


---



## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os
import sys
import subprocess
import pandas as pd

# Setup repository environment and working directory
IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")

# Load the dataset from the root directory
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Calculate metrics
total_pages = len(df)
declining_pages = (df['trend_direction'] == 'down').sum()
declining_pct = (declining_pages / total_pages) * 100
high_demand_declining = len(df[(df['impressions_90d'] >= 500) & (df['trend_direction'] == 'down')])

print(f"Total pages: {total_pages:,}")
print(f"Declining pages: {declining_pages:,} ({declining_pct:.1f}%)")
print(f"High-demand declining pages (>=500 impressions): {high_demand_declining:,}")

Total pages: 30,000
Declining pages: 16,262 (54.2%)
High-demand declining pages (>=500 impressions): 9,961


In the 30,000 total pages evaluated in the starter dataset, 16,262 pages (54.2%) show an active downward trend.

Out of those declining pages, 9,961 still receive high exposure with at least 500 impressions over 90 days.

These numbers show that over half of the catalog—and nearly 10,000 high-visibility pages—are actively losing momentum, making a machine learning prioritization queue well worth building over the next 7 weeks.

---



## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

###Answer
**What I Can Claim:**
This project can identify observed statistical patterns, directional traffic drops, and decision-support risk scores. These findings help human teams audit their content inventory more efficiently.

**What I Cannot Claim:**
This project cannot provide causal proof that editing a page guarantees traffic recovery. It also cannot predict changes to Google's proprietary search algorithm.

---



## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.